Applications/Spire/Include/Spire/Spire/RealSpinBoxModel.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef SPIRE_REAL_SPIN_BOX_MODEL_HPP
+#define SPIRE_REAL_SPIN_BOX_MODEL_HPP
+#include <QLocale>
+#include "Spire/Spire/SpinBoxModel.hpp"
+#include "Spire/Ui/RealSpinBox.hpp"
+
+namespace Spire {
+
+  //! Represents a RealSpinBox's data model.
+  class RealSpinBoxModel {
+    public:
+
+      virtual ~RealSpinBoxModel() {};
+
+      //! Returns an increment value.
+      /*
+        \param modifiers The modifiers to get the increment value of.
+      */
+      virtual RealSpinBox::Real get_increment(
+        Qt::KeyboardModifiers modifiers) const = 0;
+
+      //! Returns the initial value.
+      virtual RealSpinBox::Real get_initial() const = 0;
+
+      //! Returns the minimum acceptable value.
+      virtual RealSpinBox::Real get_minimum() const = 0;
+
+      //! Returns the maximum acceptable value.
+      virtual RealSpinBox::Real get_maximum() const = 0;
+  };
+
+  //! Wraps a SpinBoxModel to provide a generic interface.
+  template<typename T>
+  class RealSpinBoxModelImpl : public RealSpinBoxModel {
+    public:
+
+      using Type = T;
+
+      //! Constructs a RealSpinBoxModelImpl.
+      /*
+        \param model The source model.
+      */
+      RealSpinBoxModelImpl(std::shared_ptr<SpinBoxModel<Type>> model);
+
+      RealSpinBox::Real get_increment(
+        Qt::KeyboardModifiers modifiers) const override;
+
+      RealSpinBox::Real get_initial() const override;
+
+      RealSpinBox::Real get_minimum() const override;
+
+      RealSpinBox::Real get_maximum() const override;
+
+    private:
+      std::shared_ptr<SpinBoxModel<Type>> m_model;
+      CustomVariantItemDelegate m_item_delegate;
+      QLocale m_locale;
+
+      RealSpinBox::Real variant_to_real(const QVariant& value) const;
```

Comment:
Same principle as above, use ```static``` to declare members whose implementation or value does not vary with ```this```.